parser.go

```diff
@@ -2522,28 +2448,34 @@ func (p *Parser) parseOnDeleteAction() (onDelete ast.OnDeleteAction, onDeleteEnd
 	return
 }
 
-func (p *Parser) parseVectorIndexOptions() *ast.VectorIndexOptions {
-	pos := p.expectKeywordLike("OPTIONS").Pos
-	p.expect("(")
-	options := &ast.VectorIndexOptions{Options: pos}
-	for {
-		key := p.parseIdent()
-		p.expect("=")
-		value := p.parseExpr()
-		options.Records = append(options.Records, &ast.VectorIndexOption{Key: key, Value: value})
-		if p.Token.Kind == "," {
-			p.nextToken()
-			continue
-		}
-		if p.Token.Kind == ")" {
-			options.Rparen = p.Token.Pos
-			p.nextToken()
-			break
-		}
-		p.panicfAtToken(&p.Token, "expected expr or , or ), but: %s", p.Token.AsString)
+func (p *Parser) parseOptionsDef() *ast.OptionsDef {
+	key := p.parseIdent()
+	p.expect("=")
+	value := p.parseExpr()
+	return &ast.OptionsDef{
+		Name:  key,
+		Value: value,
 	}
+}
 
-	return options
+func (p *Parser) tryParseOptions() *ast.Options {
+	if !p.Token.IsKeywordLike("OPTIONS") {
+		return nil
+	}
+	return p.parseOptions()
+}
+
+func (p *Parser) parseOptions() *ast.Options {
+	optionsPos := p.expectKeywordLike("OPTIONS").Pos
+	p.expect("(")
+	optionsDefs := parseCommaSeparatedList(p, p.parseOptionsDef)
+	rparen := p.expect(")").Pos
+
+	return &ast.Options{
+		Options: optionsPos,
```

Comment:
`pos` is preferred here.

```suggestion
	pos := p.expectKeywordLike("OPTIONS").Pos
	p.expect("(")
	optionsDefs := parseCommaSeparatedList(p, p.parseOptionsDef)
	rparen := p.expect(")").Pos

	return &ast.Options{
		Options: pos,
```